ga4gh/backend.py

```diff
@@ -229,8 +245,13 @@ def _getDatasetFromVariantsRequest(self, request):
         return dataset
 
     def _getDatasetFromCompoundId(self, compoundId):
+        if compoundId is None:
+            raise exceptions.BadIdentifierException(compoundId)
+        if not hasattr(compoundId, 'split'):
```

Comment:
This is a bad sign! If we're testing for attribute existence for control flow, we should be doing things differently (except in places where the protocol is stupid, and there's no other option).